palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/repository/K8sBackingStore.java

```diff
@@ -0,0 +1,240 @@
+     * @throws IllegalArgumentException if the duration is negative
+     * @throws IllegalArgumentException if <code>key</code> is empty (once whitespace is trimmed)
+     */
+    @Override
+    public boolean add(final String key, final Class<?> valueClass,
+                       final byte[] value, final Optional<Duration> timeToLive) {
+
+        if (key == null) {
+            throw new IllegalArgumentException("key");
+        }
+        requireNonNull(valueClass, "valueClass");
+        requireNonNull(value, "value");
+        if (value.length == 0) {
+            throw new IllegalArgumentException("value is empty");
+        }
+
+        String dns1123Compatible = convertKeyToCompatible(key);
+        Resource<ConfigMap, DoneableConfigMap> configMapResource = client.configMaps().inNamespace(namespace).withName(dns1123Compatible);
+
+        if (timeToLive.isPresent()) {
+            final long epochMilli = Instant.now().plusSeconds(timeToLive.get().getSeconds()).plusNanos(timeToLive.get().getNano()).toEpochMilli();
+            ConfigMap configMap = configMapResource.createOrReplace(new ConfigMapBuilder()
+                    .withNewMetadata().withName(dns1123Compatible).endMetadata()
+                    .addToData("valueClass", valueClass.getName())
+                    .addToData("value", new String(value))
+                    .addToData("expiryTimestamp", String.valueOf(epochMilli))
+                    .build());
+            log("Upserted ConfigMap at " + configMap.getMetadata().getSelfLink() + " data " + configMap.getData());
+
+        } else {
+            ConfigMap configMap = configMapResource.createOrReplace(new ConfigMapBuilder()
+                    .withNewMetadata().withName(dns1123Compatible).endMetadata()
+                    .addToData("valueClass", valueClass.getName())
+                    .addToData("value", new String(value))
+                    .build());
+            log("Upserted ConfigMap at " + configMap.getMetadata().getSelfLink() + " data " + configMap.getData());
+        }
+        return true;
+    }
+
+
+    /**
+     * Attempt to get the given key from the backing store. Looks up the given key and attempts to get it. If the
+     * requested key couldn't be found then the value of the returned {@link SimpleCacheObject} will be empty. A new
+     * cache object should be returned for each call to this method. If the key can't be found, then the expected class
+     * should be <code>Object.class</code> inside the returned object. The metadata object in the cache object returned
+     * from a backing store should be empty.
+     *
+     * @param key the key to lookup
+     * @return a new cache object
+     * @throws IllegalArgumentException if <code>key</code> is empty (once whitespace is trimmed)
+     */
+    @Override
+    public SimpleCacheObject get(final String key) {
+        if (key == null) {
+            throw new IllegalArgumentException("key");
```

Comment:
This should throw a NullPointerException if the key is null and IllegalArg if something about the argument is wrong.

Reply:
done